In [25]:
import pandas as pd
import numpy as np

In [26]:
df=pd.read_excel("C:\\Users\\dzjc9\\Desktop\\Forex_Analysis\\bog.xlsx")

The Data df has 6 columns, the Date, the currency, currency pair,Buying, selling and Mid Rate.


Date shows the occassion or day of every transaction

Currency displays the currency which is being matched up with the Ghana Cedi

Currency pair, when an order is placed for a currency pair, the first listed currency or base currency is bought while the second listed currency in a currency pair or quote currency is sold.

The buying rate is the rate at which money dealers will buy foreign currency

The selling rate is the rate at which they will sell that currency.

The mid and mid-market rate, is the exchange rate exactly halfway between a currency's bid and ask rates.


In [27]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6479 entries, 0 to 6478
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Date           6479 non-null   object 
 1   Currency       6479 non-null   object 
 2   Currency Pair  6479 non-null   object 
 3   Buying         6479 non-null   float64
 4   Selling        6479 non-null   float64
 5   Mid Rate       6479 non-null   float64
dtypes: float64(3), object(3)
memory usage: 303.8+ KB


Our data was sourced from the Bank of Ghana Website. It contains about 6478 data points. 

The data is made collected over the span of 26 years.

On 1 July 2007, the country's existing currency, the Cedi (), was converted to the Ghana Cedi (GH), with ten thousand Cedis equaling one Ghana Cedi. The re-denomination was not expected to impact the currency's inherent worth. According to the government, "the value will remain unchanged."



In [28]:

df['Date'] = pd.to_datetime(df['Date'])
df = df.set_index(df['Date'])
#df = df.sort_index()

In [29]:
df.isnull().sum()

Date             0
Currency         0
Currency Pair    0
Buying           0
Selling          0
Mid Rate         0
dtype: int64

In [30]:
redenomination = pd.datetime(2007,7,1)

bf_red = df.loc[df['Date'] <= redenomination]
af_red = df.loc[df['Date'] > redenomination]

C:\Users\dzjc9\AppData\Local\Temp\ipykernel_13200\1619372591.py:1: FutureWarning: The pandas.datetime class is deprecated and will be removed from pandas in a future version. Import from datetime module instead.
  redenomination = pd.datetime(2007,7,1)


In [31]:
bf_red

Date   Currency Currency Pair   Buying  Selling  Mid Rate
Date                                                                      
2007-06-29 2007-06-29  US Dollar        USDGHS     0.00     0.00     0.000
2007-06-28 2007-06-28  US Dollar        USDGHS     0.00     0.00     0.000
2007-06-27 2007-06-27  US Dollar        USDGHS  9213.64  9315.82  9264.730
2007-06-26 2007-06-26  US Dollar        USDGHS     0.00     0.00     0.000
2007-06-25 2007-06-25  US Dollar        USDGHS     0.00     0.00     0.000
...               ...        ...           ...      ...      ...       ...
1996-01-08 1996-01-08  US Dollar        USDGHS  1440.91  1467.73  1454.320
1996-01-05 1996-01-05  US Dollar        USDGHS  1442.27  1469.09  1455.680
1996-01-04 1996-01-04  US Dollar        USDGHS  1438.64  1465.91  1452.275
1996-01-03 1996-01-03  US Dollar        USDGHS  1437.27  1464.36  1450.815
1996-01-02 1996-01-02  US Dollar        USDGHS  1434.09  1461.64  1447.865

[2881 rows x 6 columns]

In [32]:
af_red

Date   Currency Currency Pair  Buying  Selling  Mid Rate
Date                                                                     
2022-01-06 2022-01-06  US Dollar        USDGHS  6.0034   6.0094    6.0064
2022-01-05 2022-01-05  US Dollar        USDGHS  6.0029   6.0089    6.0059
2022-01-04 2022-01-04  US Dollar        USDGHS  6.0031   6.0091    6.0061
2021-12-31 2021-12-31  US Dollar        USDGHS  6.0031   6.0091    6.0061
2021-12-30 2021-12-30  US Dollar        USDGHS  5.9931   5.9991    5.9961
...               ...        ...           ...     ...      ...       ...
2007-07-09 2007-07-09  US Dollar        USDGHS  0.9216   0.9315    0.9266
2007-07-06 2007-07-06  US Dollar        USDGHS  0.0000   0.0000    0.0000
2007-07-05 2007-07-05  US Dollar        USDGHS  0.0000   0.0000    0.0000
2007-07-04 2007-07-04  US Dollar        USDGHS  0.0000   0.0000    0.0000
2007-07-03 2007-07-03  US Dollar        USDGHS  0.0000   0.0000    0.0000

[3598 rows x 6 columns]

In [33]:
bf_red['Buying'] =bf_red['Buying'].div(10000).round(4)
bf_red['Selling'] =bf_red['Selling'].div(10000).round(4)
bf_red['Mid Rate'] =bf_red['Mid Rate'].div(10000).round(4)

C:\Users\dzjc9\AppData\Local\Temp\ipykernel_13200\968812225.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  bf_red['Buying'] =bf_red['Buying'].div(10000).round(4)
C:\Users\dzjc9\AppData\Local\Temp\ipykernel_13200\968812225.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  bf_red['Selling'] =bf_red['Selling'].div(10000).round(4)
C:\Users\dzjc9\AppData\Local\Temp\ipykernel_13200\968812225.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row

In [34]:
bf_red

Date   Currency Currency Pair  Buying  Selling  Mid Rate
Date                                                                     
2007-06-29 2007-06-29  US Dollar        USDGHS  0.0000   0.0000    0.0000
2007-06-28 2007-06-28  US Dollar        USDGHS  0.0000   0.0000    0.0000
2007-06-27 2007-06-27  US Dollar        USDGHS  0.9214   0.9316    0.9265
2007-06-26 2007-06-26  US Dollar        USDGHS  0.0000   0.0000    0.0000
2007-06-25 2007-06-25  US Dollar        USDGHS  0.0000   0.0000    0.0000
...               ...        ...           ...     ...      ...       ...
1996-01-08 1996-01-08  US Dollar        USDGHS  0.1441   0.1468    0.1454
1996-01-05 1996-01-05  US Dollar        USDGHS  0.1442   0.1469    0.1456
1996-01-04 1996-01-04  US Dollar        USDGHS  0.1439   0.1466    0.1452
1996-01-03 1996-01-03  US Dollar        USDGHS  0.1437   0.1464    0.1451
1996-01-02 1996-01-02  US Dollar        USDGHS  0.1434   0.1462    0.1448

[2881 rows x 6 columns]

In [35]:
df1 = pd.concat([af_red,bf_red])


In [36]:
df1.head()

Date   Currency Currency Pair  Buying  Selling  Mid Rate
Date                                                                     
2022-01-06 2022-01-06  US Dollar        USDGHS  6.0034   6.0094    6.0064
2022-01-05 2022-01-05  US Dollar        USDGHS  6.0029   6.0089    6.0059
2022-01-04 2022-01-04  US Dollar        USDGHS  6.0031   6.0091    6.0061
2021-12-31 2021-12-31  US Dollar        USDGHS  6.0031   6.0091    6.0061
2021-12-30 2021-12-30  US Dollar        USDGHS  5.9931   5.9991    5.9961

In [37]:
#Resample dataframe in order to get monthly prices
df1_Monthly_Price = df1.resample('M').mean()
print(df1_Monthly_Price)

              Buying   Selling  Mid Rate
Date                                    
1996-01-31  0.145677  0.148136  0.146914
1996-02-29  0.150690  0.153045  0.151860
1996-03-31  0.154970  0.157330  0.156145
1996-04-30  0.158465  0.160545  0.159515
1996-05-31  0.160827  0.162873  0.161864
...              ...       ...       ...
2021-09-30  5.856348  5.862148  5.859248
2021-10-31  5.878795  5.884624  5.881710
2021-11-30  5.904027  5.910027  5.907027
2021-12-31  5.943890  5.949890  5.946890
2022-01-31  6.003133  6.009133  6.006133

[313 rows x 3 columns]
